utils/include/vector.h

```diff
@@ -0,0 +1,65 @@
+#ifndef VECTOR_H
+#define VECTOR_H
+
+#include <stddef.h>
+
+typedef struct {
+    size_t capacity;
+    size_t used;
+    size_t item_size;
+    char* contents;
```

Comment:
Based on the way you are accessing the vector through accessors such as `index_vector`, it appears that `contents` is meant to be a private member? If so maybe indicate this by using `_contents`

Reply:
Contents should be accessable fully as well.  index_vector just bounds checks and takes into account the item_size.

Reply:
Ok. We should just keep in mind that we want to be consistent in how we are accessing the vector, if there are accessors available and do checking for us, we may want to use those outside of this library

Reply:
Sounds good. I will add an ``index_bytes`` and ``get_contents`` function and rename ``contents`` to ``_contents``